modules/tinymce/src/plugins/table/main/ts/selection/SelectionTargets.ts

```diff
@@ -134,7 +155,17 @@ export const getSelectionTargets = (editor: Editor, selections: Selections): Sel
     onSetup(api, (targets) => isCaption(targets.element) || getClipboardData().isNone() || isDisabledFromLocked(lockedDisable));
   const onSetupMergeable = (api: UiApi) => onSetup(api, (_targets) => isDisabledForSelection('mergeable'));
   const onSetupUnmergeable = (api: UiApi) => onSetup(api, (_targets) => isDisabledForSelection('unmergeable'));
-
+  const onSetupTableWithCaption = (api: Toolbar.ToolbarToggleButtonInstanceApi) => {
+    return onSetupWithToggle(api, Fun.never, (targets) => {
+      const potentialTables = TableLookup.table(targets.element, Util.getIsRoot(editor));
+      const tableOpt = potentialTables.filter(Fun.not(Util.getIsRoot(editor)));
+
+      return tableOpt.map((table) => {
+        const captionElm = editor.dom.select('caption', table.dom)[0];
+        return Type.isNonNullable(captionElm);
+      }).getOr(false);
+    });
+  };
```

Comment:
This can also be shortened to keep with the existing code style used here. eg:
```suggestion
  const onSetupTableWithCaption = (api: UiToggleApi) => onSetupWithToggle(api, Fun.never, (targets) => {
    const tableOpt = TableLookup.table(targets.element, Util.getIsRoot(editor));
    return tableOpt.exists((table) => SelectorExists.child(table, 'caption'));
  });
```